src/grt/src/fastroute/src/maze.cpp

```diff
@@ -2447,32 +2449,25 @@ int FastRouteCore::getOverflow2D(int* maxOverflow)
 int FastRouteCore::getOverflow3D()
 {
   // get overflow
-  int overflow = 0;
   int H_overflow = 0;
   int V_overflow = 0;
-  int max_H_overflow = 0;
-  int max_V_overflow = 0;
 
   int total_usage = 0;
-
-  for (int k = 0; k < num_layers_; k++) {
-    for (const auto& [i, j] : h_used_ggrid_) {
-      total_usage += h_edges_3D_[k][i][j].usage;
-      overflow = h_edges_3D_[k][i][j].usage - h_edges_3D_[k][i][j].cap;
-
-      if (overflow > 0) {
-        H_overflow += overflow;
-        max_H_overflow = std::max(max_H_overflow, overflow);
-      }
-    }
-    for (const auto& [i, j] : v_used_ggrid_) {
-      total_usage += v_edges_3D_[k][i][j].usage;
-      overflow = v_edges_3D_[k][i][j].usage - v_edges_3D_[k][i][j].cap;
-      if (overflow > 0) {
-        V_overflow += overflow;
-        max_V_overflow = std::max(max_V_overflow, overflow);
-      }
-    }
+  int kstep = y_grid_ * x_grid_;
+  int istep = x_grid_;
+
+#pragma omp parallel for schedule(static) reduction(+ : H_overflow) \
+    reduction(+ : V_overflow) reduction(+ : total_usage)
+  for (int iter = 0; iter < num_layers_ * y_grid_ * x_grid_; iter++) {
+    int k = iter / kstep;
+    int i = (iter % kstep) / istep;
+    int j = (iter % kstep) % istep;
```

Comment:
It think it would be simpler to keep the original loop structure and apply `collapse(2)` to the prama.  The original loop was over h_used_ggrid_ rather than all points.